plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataType.java

```diff
@@ -272,6 +272,13 @@ void resolveValueTypeFromBaseType(DBRProgressMonitor monitor) {
     @Nullable
     String getConditionTypeCasting(boolean isInCondition, boolean castColumnName) {
         final String typeName = getTypeName();
+        DBSDataType dataType = DBUtils.getDataType(this);
+        if ((dataType != null) && (isInCondition)) {
+            String typeExt = CommonUtils.toString(dataType.geTypeExtension());
+            if ("E".equals(typeExt)) { // for future - maybe not only Enum? But I have no such deep knowledge
+        	return "::text";
+            }    
+        }
```

Comment:
- Unnecessary call to `getDataType` as it will yield the same reference as `this`
- Unnecessary string conversion, you can access its actual type since you're inside the PostgreSQL bundle
- Not useful comment

```suggestion
        if (isInCondition && typeCategory == PostgreTypeCategory.E) {
            return "::text";
        }
```